btk/measure.py

```diff
@@ -1,175 +1,239 @@
+                                 The pixels set to True in the i-th channel correspond to the i-th
+                                 object. The order should correspond to the order in the returned
+                                 `catalog`.
+
+Omitted keys in the returned dictionary are automatically assigned a `None` value (except for
+`catalog` which is a mandatory entry).
+"""
+import astropy
 import numpy as np
 import sep
 from skimage.feature import peak_local_max
 
 from btk.multiprocess import multiprocess
 
 
-class MeasurementParams(ABC):
-    """Class with functions to perform detection/deblending/measurement."""
-
-    def make_measurement(self, data, index):
-        """Function describing how the measurement algorithm is run.
-
-        Args:
-            data (dict): Output generated by btk.draw_blends containing blended
-                         images, isolated images, observing conditions and
-                         blend catalog, for a given batch.
-            index (int): Index number of blend scene in the batch to preform
-                         measurement on.
-
-        Returns:
-            output of measurement algorithm (fluxes, shapes, size, etc.) as
-            an astropy catalog.
-        """
-        return None
-
-    def get_deblended_images(self, data, index):
-        """Function describing how the deblending algorithm is run.
-
-        Args:
-            data (dict): Output generated by btk.draw_blends containing blended
-                         images, isolated images, observing conditions and
-                         blend catalog, for a given batch.
-            index (int): Index number of blend scene in the batch to preform
-                         measurement on.
-
-        Returns:
-            output of deblending algorithm as a dict.
-        """
-        return None
-
+def basic_measure(batch, idx):
+    """Return centers detected when object detection is performed on the
+    input image with skimage.feature.peak_local_max.
+
+    NOTE: This function does not support the multi-resolution feature.
+
+    Args:
+        batch (dict): Output of DrawBlendsGenerator object's `__next__` method.
+        idx (int): Index number of blend scene in the batch to preform
+            measurement on.
+
+    Returns:
+            dict containing catalog with entries corresponding to measured peaks.
+    """
+    if isinstance(batch["blend_images"], dict):
+        raise NotImplementedError("This function does not support the multi-resolution feature.")
+
+    image = np.mean(batch["blend_images"][idx], axis=0)
+
+    # set detection threshold to 5 times std of image
+    threshold = 5 * np.std(image)
+    coordinates = peak_local_max(image, min_distance=2, threshold_abs=threshold)
+
+    # construct catalog from measurement.
+    catalog = astropy.table.Table()
+    catalog["dx"] = coordinates[:, 1]
+    catalog["dy"] = coordinates[:, 0]
```

Comment:
Not very important but the naming `dx` and `dy` for centres is a bit puzzling.
Naturally I would expect something like `x_cen` or `x_peak`.

Reply:
Yeah I think it can be puzzling too. I like your names better.